### PR TITLE
Fix regex env removal vs. non-utf8 env entries.

### LINKED ISSUE #105: Env var removal via regex blows up when the env contains non-utf8 env vars or values
The issue is here:
https://github.com/a-scie/jump/blob/faa65b7875c1eee5fa902d81138782b6f1d00232/jump/src/process.rs#L65-L66

The `env::vars()` call effectively asserts that the env will be fully utf8 decodable. Instead the loop should be over `env::vars_os()` which works in OsString and does not try to do conversions.

### PR DESCRIPTION
**Pull request by jsirois**
Previously the scie-jump boot would crash when a manifest requested
regex removals and the ambient environment contained non-utf8 entries.

Fixes #105

### CODE REVIEW

File: jump/src/process.rs [hunk 1]
```diff
@@ -63,9 +63,20 @@ impl EnvVars {
                     removals.insert(name.to_owned());
                 }
                 EnvVar::RemoveMatching(regex) => {
-                    for (name, _) in env::vars() {
-                        if regex.is_match(name.as_str()) {
-                            removals.insert(name.into());
+                    for (name, _) in env::vars_os() {
```
Comment (jsirois) [inline jump/src/process.rs:66]: Confirmed the new unit tests fail on Windows and Linux prior to this fix and the new example IT in python/test.sh fails on Linux.

**Comment by jsirois** [comment]
I'm going to push this along shortly to get out a release and update scie-pants to it and release it in-turn.

**Review by benjyw — APPROVED** [review]

File: jump/src/process.rs [hunk 2]
```diff
@@ -63,9 +63,20 @@ impl EnvVars {
...
+                        match name.into_string() {
+                            Ok(name_utf8) => {
+                                if regex.is_match(&name_utf8) {
+                                    removals.insert(name_utf8.into());
+                                }
+                            }
+                            Err(name_os) => {
+                                warn!(
```
Comment (jsirois) [inline jump/src/process.rs:74]: This is not an awesome fix. Ideally you'd use a bytes Regex and maybe: https://crates.io/crates/os_str_bytes

Comment (jsirois) [reply jump/src/process.rs:74]: Filed #148 to address this properly.

Comment (jsirois) [reply jump/src/process.rs:74]: Fixed in #149.

**Pull request merged**